apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/MoveObjectTest.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2025 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Storage.V1.IntegrationTests.TestHelpers;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests;
+
+[Collection(nameof(StorageFixture))]
+public class MoveObjectTest
+{
+    private readonly StorageFixture _fixture;
+    private readonly string _bucket;
+    private readonly string _name1;
+    private readonly string _name2;
+    private readonly string _contentType1;
+    private readonly string _contentType2;
+    private readonly MemoryStream _source1;
+    private readonly MemoryStream _source2;
+
+    public MoveObjectTest(StorageFixture fixture)
+    {
+        _fixture = fixture;
+        _bucket = _fixture.HnsBucket;
+        _name1 = IdGenerator.FromGuid();
+        _name2 = IdGenerator.FromGuid();
+        _contentType1 = "application/octet-stream";
+        _contentType2 = "application/x-replaced";
+        _source1 = GenerateData(100);
+        _source2 = GenerateData(50);
+    }
+
+    // Moves an object within a hierarchical namespace enabled bucket.
+    [Fact]
+    public async Task MoveObjectAsync()
```

Comment:
In libraries we don't usually tests service functionality, we test library functionality. Service functionality should be tested on the service side. This is important for maintainability. 

That an object is replaced if it already existed is not library functionality: library code is not doing anything specific for it. We only need to test that, after the move, there's a `destinationName` object with the same data that was on the `originName` object and that the `originName` object does not exist anymore (to make certain that library code is calling the move operation and not the copy operation). We don't need to test what happens if there was already a `destinationName` object on the bucket, etc.

I'm proposing changes for this.